### PR TITLE
test: enforce x-auth-token removal

### PR DESCRIPTION
**Pull request by DenBeke**

### CODE REVIEW

File: internal/testhelpers/httprecorder/check_tokens.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+for file in $files
+do
+    echo "checking $file";
+    if grep -i --quiet "x-auth-token" $file
+    then
+        echo "found x-auth-token in file $file";
+        exit 1;
+    fi
+done
+exit 0;
```
Comment (QuentinBrosse) [inline internal/testhelpers/httprecorder/check_tokens.sh]: add newline

File: internal/testhelpers/httprecorder/check_tokens.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
```
Comment (QuentinBrosse) [inline internal/testhelpers/httprecorder/check_tokens.sh]: You don't think that it's a little bit strange to have a `shell` script in a go package? :/

File: internal/testhelpers/httprecorder/recorder.go [hunk 3]
```diff
@@ -41,6 +53,15 @@ func CreateRecordedScwClient(cassetteName string) (*scw.Client, *recorder.Record
 	// Add a filter which removes Authorization headers from all requests:
 	r.AddFilter(func(i *cassette.Interaction) error {
 		delete(i.Request.Headers, "x-auth-token")
+		delete(i.Request.Headers, "X-Auth-Token")
+
+		if UpdateCassette {
+			secretKey, _ := config.GetSecretKey()
+			if i != nil && strings.Contains(fmt.Sprintf("%v", *i), secretKey) {
+				panic(errors.New("found secret key in cassette"))
```
Comment (QuentinBrosse) [inline internal/testhelpers/httprecorder/recorder.go]: `fmt.Errorf` (for consistency)?

File: internal/testhelpers/httprecorder/check_tokens.sh (hunk 2 above)
Comment (DenBeke) [reply internal/testhelpers/httprecorder/check_tokens.sh]: It depends on the way you look at it. Having a script in a Go package, next to the files it is related to doesn't seem strange to me...

But tell me which other location you prefer.

File: internal/testhelpers/httprecorder/recorder.go (hunk 3 above)
Comment (DenBeke) [reply internal/testhelpers/httprecorder/recorder.go]: yeps

**Pull request merged**